src/Umbraco.Infrastructure/Services/Implement/DataTypeService.cs

```diff
@@ -281,7 +300,31 @@ public IEnumerable<IDataType> GetAll(params int[] ids)
         {
             using (var scope = ScopeProvider.CreateScope(autoComplete: true))
             {
-                return _dataTypeRepository.GetMany(ids);
+                var dataTypes = _dataTypeRepository.GetMany(ids);
+                ConvertMissingEditorsOfDataTypesToLabels(dataTypes);
+                return dataTypes;
+            }
+        }
+
+        private void ConvertMissingEditorOfDataTypeToLabel(IDataType dataType)
+        {
+            if (dataType == null)
+            {
+                return;
+            }
+
+            ConvertMissingEditorsOfDataTypesToLabels(new[] { dataType });
+        }
+
+        private void ConvertMissingEditorsOfDataTypesToLabels(IEnumerable<IDataType> dataTypes)
```

Comment:
maybe this can just be `params IDataType[] dataTypes` and then we only need the single method `ConvertMissingEditorsOfDataTypesToLabels`?